Algorithm.CSharp/OptionSplitRegressionAlgorithm.cs

```diff
@@ -0,0 +1,119 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Orders;
+using QuantConnect.Securities.Option;
+using QuantConnect.Brokerages;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This is an option split regression algorithm
+    /// </summary>
+    public class OptionSplitRegressionAlgorithm : QCAlgorithm
+    {
+        private const string UnderlyingTicker = "AAPL";
+        public readonly Symbol Underlying = QuantConnect.Symbol.Create(UnderlyingTicker, SecurityType.Equity, Market.USA);
+        public readonly Symbol OptionSymbol = QuantConnect.Symbol.Create(UnderlyingTicker, SecurityType.Option, Market.USA);
+
+        public override void Initialize()
+        {
+            // this test opens position in the first day of trading, lives through stock split (7 for 1), and closes adjusted position on the second day
+            SetStartDate(2014, 06, 06);
+            SetEndDate(2014, 06, 09);
+            SetCash(1000000);
+
+            var equity = AddEquity(UnderlyingTicker);
+            var option = AddOption(UnderlyingTicker);
+
+            equity.SetDataNormalizationMode(DataNormalizationMode.Raw);
```

Comment:
Yes, backadjusting underlying price would distort many algos that e.g. work with ATM contracts (most liquid). Those algos find ATM contracts at runtime and will find the ATM strikes at different levels in case of adjustments in backtests.